src/azure-cli/azure/cli/command_modules/cdn/_params.py

```diff
@@ -262,3 +221,305 @@ def load_arguments(self, _):
+        c.argument('rule_name', id_part=None)
+
+    # AFD Security Policy #
+    with self.argument_context('afd security-policy') as c:
+        c.argument('profile_name', help=profile_name_help, id_part='name')
+        c.argument('security_policy_name', id_part='child_name_1', help='Name of the security policy.')
+        c.argument('waf_policy', help='The ID of Front Door WAF policy.')
+        c.argument(
+            'domains',
+            nargs='+',
+            help='The domains to associate with the WAF policy. Could either be the ID of an endpoint'
+                 ' (default domain will be used in that case) or ID of a custom domain.')
+
+    with self.argument_context('afd security-policy list') as c:
+        c.argument('profile_name', id_part=None)
+        c.argument('security_policy_name', id_part=None)
+
+    # AFD Custom Domain #
+    with self.argument_context('afd custom-domain') as c:
+        c.argument('profile_name', help=profile_name_help, id_part='name')
+        c.argument('custom_domain_name', custom_name_arg_type, id_part="child_name_1",
+                   help='Name of the custom domain.')
+        c.argument('certificate_type', arg_type=get_enum_type([item.value for item in list(AfdCertificateType)]),
+                   help='Defines the source of the SSL certificate.')
+        c.argument('minimum_tls_version', arg_type=get_enum_type([item.value for item in list(AfdMinimumTlsVersion)]),
+                   help='TLS protocol version that will be used for Https.')
+        c.argument('host_name', help='The host name of the domain. Must be a domain name.')
+        c.argument('azure_dns_zone', help='ID of the Azure DNS zone.')
+        c.argument('secret', help='ID of the Azure key vault certificate.')
+
+    with self.argument_context('afd custom-domain list') as c:
+        c.argument('profile_name', id_part=None)
+        c.argument('custom_domain_name', custom_name_arg_type, id_part=None, help='Name of the custom domain.')
+
+    # AFD Secret #
+    with self.argument_context('afd secret') as c:
+        c.argument('profile_name', help=profile_name_help, id_part='name')
+        c.argument('secret_name', secret_name_arg_type, id_part="child_name_1", help='Name of the custom domain.')
+        c.argument('use_latest_version', arg_type=get_three_state_flag(),
+                   help="Whether to use the latest version for the certificate.")
+        c.argument('secret_source', help='ID of the Azure key vault certificate.')
+        c.argument('secret_version', help='Version of the certificate to be used.')
+
+    with self.argument_context('afd secret list') as c:
+        c.argument('profile_name', id_part=None)
+        c.argument('secret_name', id_part=None)
+
+    # AFD Log Analytic #
+    with self.argument_context('afd log-analytic') as c:
+        c.argument('group_by', nargs='+',
+                   arg_type=get_enum_type(["httpStatusCode", "protocol", "cacheStatus", "country", "customDomain"]),
```

Comment:
Is the enum value defined in SDK? If so you can import the enum class here.

Reply:
Not yet. This version of swagger definition the SDK is based has some mistakes which omit these enum definition. I will fix it once we have the new SDK generated.